### ISSUE TITLE
Code Coverage Html file not generated for Unit Tests (Yii 2)

### ISSUE THREAD

**Issue by jmbantay** [Codeception/Codeception#5655]
#### What are you trying to achieve?
Generate html code coverage for unit tests
```bash
codecept run unit --coverage-html
```

#### What do you get instead?
```bash
[yii\base\ErrorException]                                                   
count(): Parameter must be an array or an object that implements Countable
```
No html coverage file was generated. --coverage-text and --coverage-xml both work perfectly.

The unit tests work. Running the tests with --coverage produces the following output:
```bash
Unit Tests (3) -----------------------------------------------------------------
Test search no params (ExperimentDataTest::testSearchNoParams)            Ok
Test rules (ExperimentDataTest::testRules)                                Ok
Test search with params (ExperimentDataTest::testSearchWithParams)        Ok.. 
--------------------------------------------------------------------------------
```
I also have acceptance tests and when i run:
```
codecept run acceptance --coverage-html
```
it succesfully creates the html coverage report.

### Details

* Codeception version: 2.1.11
* PHP Version: 7.3
* Operating System: Ubuntu 16.04
* Installation type: Composer
* List of installed packages (`composer show`)
```
behat/gherkin                      v4.4.5  Gherkin DSL parser for PHP 5.3
bluezed/yii2-scroll-top            v1.0.4  This Yii 2 widget adds a simple ...
bower-asset/bootstrap              v3.4.1 
bower-asset/inputmask              3.3.11 
bower-asset/jquery                 3.4.1  
bower-asset/jquery-ui              1.12.1 
bower-asset/materialize            1.0.0  
bower-asset/punycode               v1.3.2 
bower-asset/yii2-pjax              2.0.7.1
cebe/markdown                      1.2.1   A super fast, highly extensible ...
codeception/base                   2.3.9   BDD-style testing framework
codeception/specify                0.4.6   BDD code blocks for PHPUnit and ...
codeception/stub                   1.0.4   Flexible Stub wrapper for PHPUni...
codeception/verify                 0.4.0   BDD assertion library for PHPUnit
doctrine/instantiator              1.2.0   A small, lightweight utility to ...
doctrine/lexer                     1.1.0   PHP Doctrine Lexer parser librar...
egulias/email-validator            2.1.11  A library for validating emails ...
ezyang/htmlpurifier                v4.11.0 Standards compliant HTML filter ...
fzaninotto/faker                   v1.8.0  Faker is a PHP library that gene...
guzzlehttp/guzzle                  6.3.3   Guzzle is a PHP HTTP client library
guzzlehttp/promises                v1.3.1  Guzzle promises library
guzzlehttp/psr7                    1.6.1   PSR-7 message implementation tha...
macgyer/yii2-materializecss        3.0.0   Materialize CSS implementation f...
myclabs/deep-copy                  1.9.3   Create deep copies (clones) of y...
opis/closure                       3.3.1   A library that can be used to se...
phar-io/manifest                   1.0.1   Component for reading phar.io ma...
phar-io/version                    1.0.1   Library for handling version inf...
phpdocumentor/reflection-common    1.0.1   Common reflection classes used b...
phpdocumentor/reflection-docblock  4.3.1   With this component, a library c...
phpdocumentor/type-resolver        0.4.0  
phpspec/php-diff                   v1.1.0  A comprehensive library for gene...
phpspec/prophecy                   1.8.1   Highly opinionated mocking frame...
phpunit/php-code-coverage          5.3.2   Library that provides collection...
phpunit/php-file-iterator          1.4.5   FilterIterator implementation th...
phpunit/php-text-template          1.2.1   Simple template engine.
phpunit/php-timer                  1.0.9   Utility class for timing
phpunit/php-token-stream           2.0.2   Wrapper around PHP's tokenizer e...
phpunit/phpunit                    6.5.14  The PHP Unit Testing framework.
phpunit/phpunit-mock-objects       5.0.10  Mock Object library for PHPUnit
psr/container                      1.0.0   Common Container Interface (PHP ...
psr/http-message                   1.0.1   Common interface for HTTP messages
ralouphie/getallheaders            3.0.3   A polyfill for getallheaders.
sebastian/code-unit-reverse-lookup 1.0.1   Looks up which function or metho...
sebastian/comparator               2.1.3   Provides the functionality to co...
sebastian/diff                     2.0.1   Diff implementation
sebastian/environment              3.1.0   Provides functionality to handle...
sebastian/exporter                 3.1.1   Provides the functionality to ex...
sebastian/global-state             2.0.0   Snapshotting of global state
sebastian/object-enumerator        3.0.3   Traverses array structures and o...
sebastian/object-reflector         1.1.1   Allows reflection of object attr...
sebastian/recursion-context        3.0.0   Provides functionality to recurs...
sebastian/resource-operations      1.0.0   Provides a list of PHP built-in ...
sebastian/version                  2.0.1   Library that helps with managing...
swiftmailer/swiftmailer            v6.2.1  Swiftmailer, free feature-rich P...
symfony/browser-kit                v4.2.4  Symfony BrowserKit Component
symfony/console                    v4.3.3  Symfony Console Component
symfony/css-selector               v4.3.3  Symfony CssSelector Component
symfony/dom-crawler                v4.3.3  Symfony DomCrawler Component
symfony/event-dispatcher           v4.3.3  Symfony EventDispatcher Component
symfony/event-dispatcher-contracts v1.1.5  Generic abstractions related to ...
symfony/finder                     v4.3.3  Symfony Finder Component
symfony/polyfill-ctype             v1.12.0 Symfony polyfill for ctype funct...
symfony/polyfill-iconv             v1.12.0 Symfony polyfill for the Iconv e...
symfony/polyfill-intl-idn          v1.12.0 Symfony polyfill for intl's idn_...
symfony/polyfill-mbstring          v1.12.0 Symfony polyfill for the Mbstrin...
symfony/polyfill-php72             v1.12.0 Symfony polyfill backporting som...
symfony/polyfill-php73             v1.12.0 Symfony polyfill backporting som...
symfony/service-contracts          v1.1.5  Generic abstractions related to ...
symfony/yaml                       v4.3.3  Symfony Yaml Component
theseer/tokenizer                  1.1.3   A small library for converting t...
webmozart/assert                   1.4.0   Assertions to validate method in...
yiisoft/yii2                       2.0.25  Yii PHP Framework Version 2
yiisoft/yii2-bootstrap             2.0.10  The Twitter Bootstrap extension ...
yiisoft/yii2-composer              2.0.8   The composer plugin for Yii exte...
yiisoft/yii2-debug                 2.1.7   The debugger extension for the Y...
yiisoft/yii2-faker                 2.0.4   Fixture generator. The Faker int...
yiisoft/yii2-gii                   2.1.1   The Gii extension for the Yii fr...
yiisoft/yii2-jui                   2.0.7   The Jquery UI extension for the ...
yiisoft/yii2-swiftmailer           2.1.2   The SwiftMailer integration for ...
```

* Suite configuration:

```yml
actor: Tester
paths:
    tests: tests
    log: tests/_output
    data: tests/_data
    support: tests/_support
    envs: tests/_envs
settings:
    bootstrap: _bootstrap.php
    colors: true
    memory_limit: 1024M
coverage:
    enabled: true
    high_limit: 60
    include: 
        - models/ExperimentData.php
extensions:
    enabled:
        - Codeception\Extension\RunFailed
modules:
    config:
        Db:
            dsn: ''
            user: ''
            password: ''
            dump: tests/_data/dump.sql
```

**Comment by Naktibalda** [comment]
You get `count(): Parameter must be an array or an object that implements Countable` because you are running code which isn't ready for PHP 7.3

Codeception 2.1 is long dead, please upgrade to Codeception 3.0 and see if code coverage works with it.

**Comment by jmbantay** [comment]
I tried to install Codeception 3.0. However, I get the following error:
```
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Conclusion: don't install codeception/codeception 3.0.3
    - Conclusion: don't install codeception/codeception 3.0.2
    - Conclusion: don't install codeception/codeception 3.0.1
    - Installation request for codeception/codeception ^3.0 -> satisfiable by codeception/codeception[3.0.0, 3.0.1, 3.0.2, 3.0.3].
    - Conclusion: remove codeception/stub 1.0.4
    - codeception/codeception 3.0.0 requires codeception/stub ^2.0 -> satisfiable by codeception/stub[2.0.0, 2.0.1, 2.0.4, 2.1.0].
    - Can only install one of: codeception/stub[2.0.0, 1.0.4].
    - Can only install one of: codeception/stub[2.0.1, 1.0.4].
    - Can only install one of: codeception/stub[2.0.4, 1.0.4].
    - Can only install one of: codeception/stub[2.1.0, 1.0.4].
    - Installation request for codeception/stub (installed at 1.0.4) -> satisfiable by codeception/stub[1.0.4].


Installation failed, reverting ./composer.json to its original content.
```

I used this to install:
```
composer require "codeception/codeception" --dev
```

and also tried:
```
composer update "codeception/codeception" --dev
```

what could be the problem here?

**Comment by jmbantay** [comment]
Scratch my last comment. I was able to install Codeception 3.0.3 by deleting my project's vendor folder, removing all codeception-related entries in the composer.json file and deleting composer.lock

This is what i have when i run ```composer show```
```
behat/gherkin                      v4.6.0     Gherkin DSL parser for PHP 5.3
bluezed/yii2-scroll-top            v1.0.4     This Yii 2 widget adds a simp...
bower-asset/bootstrap              v3.4.1    
bower-asset/inputmask              3.3.11    
bower-asset/jquery                 3.4.1     
bower-asset/jquery-ui              1.12.1    
bower-asset/materialize            1.0.0     
bower-asset/punycode               v1.3.2    
bower-asset/yii2-pjax              2.0.7.1   
cebe/markdown                      1.2.1      A super fast, highly extensib...
codeception/codeception            3.0.3      BDD-style testing framework
codeception/phpunit-wrapper        8.0.4      PHPUnit classes used by Codec...
codeception/stub                   2.1.0      Flexible Stub wrapper for PHP...
doctrine/instantiator              1.2.0      A small, lightweight utility ...
doctrine/lexer                     1.1.0      PHP Doctrine Lexer parser lib...
egulias/email-validator            2.1.11     A library for validating emai...
ezyang/htmlpurifier                v4.11.0    Standards compliant HTML filt...
facebook/webdriver                 1.7.1      A PHP client for Selenium Web...
fzaninotto/faker                   v1.8.0     Faker is a PHP library that g...
guzzlehttp/guzzle                  6.3.3      Guzzle is a PHP HTTP client l...
guzzlehttp/promises                v1.3.1     Guzzle promises library
guzzlehttp/psr7                    1.6.1      PSR-7 message implementation ...
hoa/consistency                    1.17.05.02 The Hoa\Consistency library.
hoa/console                        3.17.05.02 The Hoa\Console library.
hoa/event                          1.17.01.13 The Hoa\Event library.
hoa/exception                      1.17.01.16 The Hoa\Exception library.
hoa/file                           1.17.07.11 The Hoa\File library.
hoa/iterator                       2.17.01.10 The Hoa\Iterator library.
hoa/protocol                       1.17.01.14 The Hoa\Protocol library.
hoa/stream                         1.17.02.21 The Hoa\Stream library.
hoa/ustring                        4.17.01.16 The Hoa\Ustring library.
macgyer/yii2-materializecss        3.0.0      Materialize CSS implementatio...
myclabs/deep-copy                  1.9.3      Create deep copies (clones) o...
opis/closure                       3.3.1      A library that can be used to...
phar-io/manifest                   1.0.3      Component for reading phar.io...
phar-io/version                    2.0.1      Library for handling version ...
phpdocumentor/reflection-common    1.0.1      Common reflection classes use...
phpdocumentor/reflection-docblock  4.3.1      With this component, a librar...
phpdocumentor/type-resolver        0.4.0     
phpspec/php-diff                   v1.1.0     A comprehensive library for g...
phpspec/prophecy                   1.8.1      Highly opinionated mocking fr...
phpunit/php-code-coverage          7.0.7      Library that provides collect...
phpunit/php-file-iterator          2.0.2      FilterIterator implementation...
phpunit/php-text-template          1.2.1      Simple template engine.
phpunit/php-timer                  2.1.2      Utility class for timing
phpunit/php-token-stream           3.1.0      Wrapper around PHP's tokenize...
phpunit/phpunit                    8.3.4      The PHP Unit Testing framework.
psr/container                      1.0.0      Common Container Interface (P...
psr/http-message                   1.0.1      Common interface for HTTP mes...
ralouphie/getallheaders            3.0.3      A polyfill for getallheaders.
sebastian/code-unit-reverse-lookup 1.0.1      Looks up which function or me...
sebastian/comparator               3.0.2      Provides the functionality to...
sebastian/diff                     3.0.2      Diff implementation
sebastian/environment              4.2.2      Provides functionality to han...
sebastian/exporter                 3.1.1      Provides the functionality to...
sebastian/global-state             3.0.0      Snapshotting of global state
sebastian/object-enumerator        3.0.3      Traverses array structures an...
sebastian/object-reflector         1.1.1      Allows reflection of object a...
sebastian/recursion-context        3.0.0      Provides functionality to rec...
sebastian/resource-operations      2.0.1      Provides a list of PHP built-...
sebastian/type                     1.1.3      Collection of value objects t...
sebastian/version                  2.0.1      Library that helps with manag...
swiftmailer/swiftmailer            v6.2.1     Swiftmailer, free feature-ric...
symfony/browser-kit                v4.2.4     Symfony BrowserKit Component
symfony/console                    v4.3.3     Symfony Console Component
symfony/css-selector               v4.3.3     Symfony CssSelector Component
symfony/dom-crawler                v4.3.3     Symfony DomCrawler Component
symfony/event-dispatcher           v4.3.3     Symfony EventDispatcher Compo...
symfony/event-dispatcher-contracts v1.1.5     Generic abstractions related ...
symfony/finder                     v4.3.3     Symfony Finder Component
symfony/polyfill-ctype             v1.12.0    Symfony polyfill for ctype fu...
symfony/polyfill-iconv             v1.12.0    Symfony polyfill for the Icon...
symfony/polyfill-intl-idn          v1.12.0    Symfony polyfill for intl's i...
symfony/polyfill-mbstring          v1.12.0    Symfony polyfill for the Mbst...
symfony/polyfill-php72             v1.12.0    Symfony polyfill backporting ...
symfony/polyfill-php73             v1.12.0    Symfony polyfill backporting ...
symfony/process                    v4.3.3     Symfony Process Component
symfony/service-contracts          v1.1.5     Generic abstractions related ...
symfony/yaml                       v4.3.3     Symfony Yaml Component
theseer/tokenizer                  1.1.3      A small library for convertin...
webmozart/assert                   1.4.0      Assertions to validate method...
yiisoft/yii2                       2.0.25     Yii PHP Framework Version 2
yiisoft/yii2-bootstrap             2.0.10     The Twitter Bootstrap extensi...
yiisoft/yii2-composer              2.0.8      The composer plugin for Yii e...
yiisoft/yii2-debug                 2.1.7      The debugger extension for th...
yiisoft/yii2-faker                 2.0.4      Fixture generator. The Faker ...
yiisoft/yii2-gii                   2.1.1      The Gii extension for the Yii...
yiisoft/yii2-jui                   2.0.7      The Jquery UI extension for t...
yiisoft/yii2-swiftmailer           2.1.2      The SwiftMailer integration f...
```

However, the original error persists.
```
[yii\base\ErrorException]                                                   
count(): Parameter must be an array or an object that implements Countable
```

**Comment by Naktibalda** [comment]
Please run codeception with `-vvv` and show the full stacktrace of that exception.

**Comment by jmbantay** [comment]
Hi, sorry I took long to respond. Here's  the stacktrace

```
Exception trace:
 () at /home/jaymar/.composer/vendor/phpunit/php-code-coverage/src/CodeCoverage/Report/HTML/Renderer/File.php:294
 PHP_CodeCoverage_Report_HTML_Renderer_File->render() at /home/jaymar/.composer/vendor/phpunit/php-code-coverage/src/CodeCoverage/Report/HTML.php:121
 PHP_CodeCoverage_Report_HTML->process() at /home/jaymar/.composer/vendor/codeception/codeception/src/Codeception/Coverage/Subscriber/Printer.php:103
 Codeception\Coverage\Subscriber\Printer->printHtml() at /home/jaymar/.composer/vendor/codeception/codeception/src/Codeception/Coverage/Subscriber/Printer.php:68
 Codeception\Coverage\Subscriber\Printer->printResult() at /home/jaymar/.composer/vendor/symfony/event-dispatcher/EventDispatcher.php:174
 ::call_user_func:{/home/jaymar/.composer/vendor/symfony/event-dispatcher/EventDispatcher.php:174}() at /home/jaymar/.composer/vendor/symfony/event-dispatcher/EventDispatcher.php:174
 Symfony\Component\EventDispatcher\EventDispatcher->doDispatch() at /home/jaymar/.composer/vendor/symfony/event-dispatcher/EventDispatcher.php:43
 Symfony\Component\EventDispatcher\EventDispatcher->dispatch() at /home/jaymar/.composer/vendor/codeception/codeception/src/Codeception/Codecept.php:224
 Codeception\Codecept->printResult() at /home/jaymar/.composer/vendor/codeception/codeception/src/Codeception/Command/Run.php:258
 Codeception\Command\Run->execute() at /home/jaymar/.composer/vendor/symfony/console/Command/Command.php:261
 Codeception\Command\Run->run() at /home/jaymar/.composer/vendor/symfony/console/Application.php:817
 Symfony\Component\Console\Application->doRunCommand() at /home/jaymar/.composer/vendor/symfony/console/Application.php:185
 Symfony\Component\Console\Application->doRun() at /home/jaymar/.composer/vendor/symfony/console/Application.php:116
 Symfony\Component\Console\Application->run() at /home/jaymar/.composer/vendor/codeception/codeception/codecept:28

```

**Comment by jmbantay** [comment]
I have decided to just create another basic Yii 2 app and transfer necessary files there. Reinstalled everything. It works now. 

To be honest, I only started coding in PHP 3 weeks ago and my installation was messy. Now that I have a tighter grasp of PHP and Yii2 Framework, I was able to actually run my tests and produce an html coverage file without any errors. I can't figure out what exactly went wrong, but it was surely my carelessness that caused this.

Thanks anyway for trying to help @Naktibalda, I appreciate your efforts.
